# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1546199936 28800
#      Sun Dec 30 11:58:56 2018 -0800
# Branch stable
# Node ID 3c2264556abc07e8c3a364ae673aa22be488b0bd
# Parent  74368b4f9d8c28e9cd2f892c865b15c2606487b6
gui: exit even when shadowing the 'exit' or 'quit' builtins (bug #55324)

* main-window.cc (main_window::closeEvent): Call Fquit builtin instead of
evaluating the name "exit".

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -1656,17 +1656,17 @@ namespace octave
     // GUI to be updated.
 
     octave_link::post_event (this, &main_window::refresh_workspace_callback);
   }
 
   void main_window::closeEvent (QCloseEvent *e)
   {
     e->ignore ();
-    octave_cmd_exec *cmd = new octave_cmd_exec ("exit");
+    octave_cmd_builtin *cmd = new octave_cmd_builtin (&Fquit, ovl ());
     queue_cmd (cmd);
   }
 
   void main_window::construct_central_widget (void)
   {
     // Create and set the central widget.  QMainWindow takes ownership of
     // the widget (pointer) so there is no need to delete the object upon
     // destroying this main_window.
